### DISCUSSION TITLE
Commits w/ same change ID

### DISCUSSION THREAD

**Discussion by crepererum** [jj-vcs/jj#4193] [Q&A]
In the recent days I ran into situations where jj got confused about the commit<>change mapping. This was visible in `jj log`, e.g.

![image](https://github.com/user-attachments/assets/a7e26f51-6079-4e29-9e20-a97d35f5a860)

(note the red coloring of the change ID and the two question marks `??`)

Using that commit or an affected branch results in

> Some of these commits have the same change id.

jj will also list the affected commits so you can abandon the ones you don't want. However it's rather tricky to find correct/wanted commit, you basically have to check them out and manually judge which is the "later" one (in terms of your personal work progress / wall clock).

I am trying to figure out what I am doing "wrong" to trigger this issue and I have a rough idea:

I'm using jj with GitHub. The target repo has a quirk that sometimes a CI bot will push to your branches to "fix" certain issues it found (it adds a commit on top). I am wondering if this trips jj.

Any idea? Does anyone else run into this problem? I couldn't find any other reports.

**Comment by PhilipMetzger** [comment]
These are divergent commits see https://github.com/martinvonz/jj/blob/main/docs/FAQ.md#how-do-i-deal-with-divergent-changes--after-the-change-id, because your bot writes a new history for these commits.

**Reply by crepererum** [reply]
The bot only adds commit AFTER the pushed changes (i.e. as parents), it never rewrites any existing ones. So I would understand if this leads to conflicting branches, but I don't see how this trips the change tracking.

**Comment by martinvonz** [comment]
You can determine exactly which operation caused the divergent change by running `jj log -r vmn --at-op <op id>` for various recent operations (found in `jj op log`).

**Reply by crepererum** [reply]
Thanks, I'll try next time I encounter this issue 🙂
